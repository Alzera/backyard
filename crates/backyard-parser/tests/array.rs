use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "array(1, 2);").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn short() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "[1, 2];").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn with_key() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "['a' => 1, 'b' => 2];").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn complex() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(
    &arena,
    "$a = [$key = (is_int($key) ? $value : $key) => $value === true ? $key : $value];"
  ).unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}
