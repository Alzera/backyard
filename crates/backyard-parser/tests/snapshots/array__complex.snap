---
source: crates/backyard-parser/tests/array.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: assignment
    left:
      node_type: variable
      name:
        node_type: identifier
        name:
          - 97
        loc:
          start:
            line: 1
            column: 0
            offset: 0
          end:
            line: 1
            column: 1
            offset: 1
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 0
          offset: 0
        end:
          line: 1
          column: 1
          offset: 1
      leadings: ~
      trailings: ~
    operator: Default
    right:
      node_type: array
      is_short: true
      items:
        - node_type: array_item
          key:
            node_type: assignment
            left:
              node_type: variable
              name:
                node_type: identifier
                name:
                  - 107
                  - 101
                  - 121
                loc:
                  start:
                    line: 1
                    column: 6
                    offset: 6
                  end:
                    line: 1
                    column: 9
                    offset: 9
                leadings: ~
                trailings: ~
              loc:
                start:
                  line: 1
                  column: 6
                  offset: 6
                end:
                  line: 1
                  column: 9
                  offset: 9
              leadings: ~
              trailings: ~
            operator: Default
            right:
              node_type: parenthesis
              statement:
                node_type: ternary
                condition:
                  node_type: call
                  name:
                    node_type: identifier
                    name:
                      - 105
                      - 115
                      - 95
                      - 105
                      - 110
                      - 116
                    loc:
                      start:
                        line: 1
                        column: 14
                        offset: 14
                      end:
                        line: 1
                        column: 20
                        offset: 20
                    leadings: ~
                    trailings: ~
                  arguments:
                    - node_type: argument
                      name: ~
                      value:
                        node_type: variable
                        name:
                          node_type: identifier
                          name:
                            - 107
                            - 101
                            - 121
                          loc:
                            start:
                              line: 1
                              column: 21
                              offset: 21
                            end:
                              line: 1
                              column: 24
                              offset: 24
                          leadings: ~
                          trailings: ~
                        loc:
                          start:
                            line: 1
                            column: 21
                            offset: 21
                          end:
                            line: 1
                            column: 24
                            offset: 24
                        leadings: ~
                        trailings: ~
                      loc:
                        start:
                          line: 1
                          column: 21
                          offset: 21
                        end:
                          line: 1
                          column: 21
                          offset: 21
                      leadings: ~
                      trailings: ~
                  loc:
                    start:
                      line: 1
                      column: 20
                      offset: 20
                    end:
                      line: 1
                      column: 25
                      offset: 25
                  leadings: ~
                  trailings: ~
                valid:
                  node_type: variable
                  name:
                    node_type: identifier
                    name:
                      - 118
                      - 97
                      - 108
                      - 117
                      - 101
                    loc:
                      start:
                        line: 1
                        column: 29
                        offset: 29
                      end:
                        line: 1
                        column: 34
                        offset: 34
                    leadings: ~
                    trailings: ~
                  loc:
                    start:
                      line: 1
                      column: 29
                      offset: 29
                    end:
                      line: 1
                      column: 34
                      offset: 34
                  leadings: ~
                  trailings: ~
                invalid:
                  node_type: variable
                  name:
                    node_type: identifier
                    name:
                      - 107
                      - 101
                      - 121
                    loc:
                      start:
                        line: 1
                        column: 38
                        offset: 38
                      end:
                        line: 1
                        column: 41
                        offset: 41
                    leadings: ~
                    trailings: ~
                  loc:
                    start:
                      line: 1
                      column: 38
                      offset: 38
                    end:
                      line: 1
                      column: 41
                      offset: 41
                  leadings: ~
                  trailings: ~
                loc:
                  start:
                    line: 1
                    column: 27
                    offset: 27
                  end:
                    line: 1
                    column: 38
                    offset: 38
                leadings: ~
                trailings: ~
              loc:
                start:
                  line: 1
                  column: 13
                  offset: 13
                end:
                  line: 1
                  column: 42
                  offset: 42
              leadings: ~
              trailings: ~
            loc:
              start:
                line: 1
                column: 11
                offset: 11
              end:
                line: 1
                column: 42
                offset: 42
            leadings: ~
            trailings: ~
          value:
            node_type: bin
            left:
              node_type: variable
              name:
                node_type: identifier
                name:
                  - 118
                  - 97
                  - 108
                  - 117
                  - 101
                loc:
                  start:
                    line: 1
                    column: 47
                    offset: 47
                  end:
                    line: 1
                    column: 52
                    offset: 52
                leadings: ~
                trailings: ~
              loc:
                start:
                  line: 1
                  column: 47
                  offset: 47
                end:
                  line: 1
                  column: 52
                  offset: 52
              leadings: ~
              trailings: ~
            operator: IsIdentical
            right:
              node_type: ternary
              condition:
                node_type: boolean
                is_true: true
                loc:
                  start:
                    line: 1
                    column: 58
                    offset: 58
                  end:
                    line: 1
                    column: 58
                    offset: 58
                leadings: ~
                trailings: ~
              valid:
                node_type: variable
                name:
                  node_type: identifier
                  name:
                    - 107
                    - 101
                    - 121
                  loc:
                    start:
                      line: 1
                      column: 65
                      offset: 65
                    end:
                      line: 1
                      column: 68
                      offset: 68
                  leadings: ~
                  trailings: ~
                loc:
                  start:
                    line: 1
                    column: 65
                    offset: 65
                  end:
                    line: 1
                    column: 68
                    offset: 68
                leadings: ~
                trailings: ~
              invalid:
                node_type: variable
                name:
                  node_type: identifier
                  name:
                    - 118
                    - 97
                    - 108
                    - 117
                    - 101
                  loc:
                    start:
                      line: 1
                      column: 72
                      offset: 72
                    end:
                      line: 1
                      column: 77
                      offset: 77
                  leadings: ~
                  trailings: ~
                loc:
                  start:
                    line: 1
                    column: 72
                    offset: 72
                  end:
                    line: 1
                    column: 77
                    offset: 77
                leadings: ~
                trailings: ~
              loc:
                start:
                  line: 1
                  column: 63
                  offset: 63
                end:
                  line: 1
                  column: 72
                  offset: 72
              leadings: ~
              trailings: ~
            loc:
              start:
                line: 1
                column: 54
                offset: 54
              end:
                line: 1
                column: 72
                offset: 72
            leadings: ~
            trailings: ~
          loc:
            start:
              line: 1
              column: 44
              offset: 44
            end:
              line: 1
              column: 72
              offset: 72
          leadings: ~
          trailings: ~
      loc:
        start:
          line: 1
          column: 5
          offset: 5
        end:
          line: 1
          column: 78
          offset: 78
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 3
        offset: 3
      end:
        line: 1
        column: 78
        offset: 78
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 79
    offset: 79
leadings: ~
trailings: ~
